newIDE/app/src/MainFrame/EditorContainers/HomePage/TeamSection/index.js

```diff
@@ -0,0 +1,361 @@
+        ],
+        group: { id: 'none', name: 'none' },
+      };
+      return;
+    }
+    const group = groups.find(group => group.id === memberGroups[0]);
+    if (!group) return;
+    const item = membersByGroupId[group.id];
+    if (item) {
+      item.members = [...item.members, member];
+    } else {
+      membersByGroupId[group.id] = { group, members: [member] };
+    }
+  });
+  groups.forEach(group => {
+    if (!(group.id in membersByGroupId)) {
+      membersByGroupId[group.id] = { group, members: [] };
+    }
+  });
+  return membersByGroupId;
+};
+
+const TeamSection = React.forwardRef<Props, TeamSectionInterface>(
+  ({ project, onOpenRecentFile, storageProviders }, ref) => {
+    const {
+      groups,
+      members,
+      memberships,
+      onChangeGroupName,
+      onChangeUserGroup,
+      onListUserProjects,
+      onDeleteGroup,
+      onCreateGroup,
+    } = React.useContext(TeamContext);
+    const gdevelopTheme = React.useContext(GDevelopThemeContext);
+    const forceUpdate = useForceUpdate();
+    const windowWidth = useResponsiveWindowWidth();
+    const isMobile = windowWidth === 'small';
+
+    React.useImperativeHandle(ref, () => ({
+      forceUpdate,
+    }));
+
+    const draggedUserRef = React.useRef<?User>(null);
+    const [selectedUser, setSelectedUser] = React.useState<?User>(null);
+    const [
+      selectedUserProjects,
+      setSelectedUserProjects,
+    ] = React.useState<?Array<CloudProjectWithUserAccessInfo>>(null);
+    const [
+      isLoadingUserProjects,
+      setIsLoadingUserProjects,
+    ] = React.useState<boolean>(false);
+    const [
+      showNewGroupNameField,
+      setShowNewGroupNameField,
+    ] = React.useState<boolean>(false);
+
+    const setDraggedUser = React.useCallback((user: User) => {
+      draggedUserRef.current = user;
+    }, []);
+
+    const listUserProjects = React.useCallback(
+      async (user: User) => {
+        setIsLoadingUserProjects(true);
+        try {
+          setSelectedUser(user);
+          const userProjects = await onListUserProjects(user);
+          setSelectedUserProjects(userProjects);
+        } catch (error) {
+          console.error(error);
+        } finally {
+          setIsLoadingUserProjects(false);
+        }
+      },
+      [onListUserProjects]
+    );
+
+    const membersByGroupId = groupMembersByGroupId({
+      groups,
+      members,
+      memberships,
+    });
+    if (!membersByGroupId) {
+      return (
+        <>
+          <SectionContainer title={<Trans>Team</Trans>}>
+            <SectionRow>
+              <Line>
+                <Column noMargin expand alignItems="center">
+                  <CircularProgress />
+                </Column>
+              </Line>
+            </SectionRow>
+          </SectionContainer>
+        </>
+      );
+    }
+
+    if (selectedUser && selectedUserProjects) {
+      const fileMetadataAndStorageProviderNames = transformCloudProjectsIntoFileMetadataWithStorageProviderName(
```

Comment:
I have a feeling this could almost be a separate component

Reply:
Good idea, let's extract it